### PR TITLE
feat(write-prd): executive summary + Definition of Done in every PRD

### PR DESCRIPTION
**Pull request by queso**
## What

The `write-prd` skill now opens every generated PRD with two top-matter sections, present in all tiers (Quick/Standard/Deep):

- **Executive Summary** — 2–4 sentences (scaled to tier) directly under the header: what this delivers, for whom, why now. Drafted last, placed first — it's the section the user reviews to confirm intent was captured.
- **Definition of Done** — a checklist of user-visible, observable outcomes, placed directly under the summary. **Scaffolded blank** in the pipeline flow (Face rolls the per-item acceptance criteria into it during planning and the human blesses it at the refinement gate); filled in only for a standalone PRD no mission will decompose.

## Why

Two motivations, both from recent work:
- The **execution-stage PRD (010)** established that the DoD belongs *where the reader actually looks* — under the exec summary — as the two-minute intent check that catches spec mismatch before build. This bakes that placement into every PRD the skill produces.
- A consistent summary-first shape makes PRDs skimmable and gives the *ratify-don't-author* review loop (see the async-intake PRD) a fixed place to land.

## Design choices

- Both sections are **unnumbered top-matter**, not new numbered sections — so sections 1–11 and every tier-guide reference (`Sections 1-6 + 10`, `All 11 sections`) stay intact.
- The DoD is **blank by default** in the A(i)-Team flow rather than author-filled, because an authored list would just be overwritten by Face's rollup.

## Files

- `skills/write-prd/references/prd-template.md` — the two sections + tier-guide note
- `skills/write-prd/SKILL.md` — Step 4 write guidance + Step 5 review checklist
- `skills/write-prd/references/prd-best-practices.md` — two quality-checklist items

Docs/skill-only; no runtime code touched.

## Summary by CodeRabbit

* **Documentation**
  * PRD guidance now requires an Executive Summary and Definition of Done in every tier.
  * Added recommendations for tailoring summary length and completion criteria to the PRD type.
  * Added review checklist items to confirm both sections are included.
  * Pipeline-managed PRDs may leave the Definition of Done blank, while standalone PRDs should include observable outcomes.

* **Workflow Improvements**
  * Planning reports now document ADR candidates and outcomes.
  * Updated agent configuration to improve planning efficiency.

* **Testing**
  * Added guidance for testing cancellation, dismissal, and abort actions during asynchronous operations.

### CODE REVIEW

File: skills/write-prd/SKILL.md [hunk 1]
```diff
@@ -103,6 +108,8 @@ Consult `references/prd-template.md` for the document structure.
 
 After writing the first draft, review it against this checklist:
 
+- Is there an Executive Summary at the top that captures what/who/why in a few sentences?
+- Is there a Definition of Done section directly under it — blank scaffold (pipeline PRD) or filled with observable, user-visible outcomes (standalone PRD)?
```
[the remark this answers is not in the document]
Comment (queso) [reply skills/write-prd/SKILL.md]: Fixed in `c29934b`. Both quality gates now pin the exact unnumbered headings *and* their placement, rather than just presence and adjacency:

- `skills/write-prd/SKILL.md` (Step 5) — requires an unnumbered `## Executive Summary` (explicitly not `## 1. Executive Summary`) above every numbered section, with `## Definition of Done` immediately below it and still ahead of `## 1.`
- `skills/write-prd/references/prd-best-practices.md` (quality checklist) — same structural check.

So a PRD with numbered top-matter no longer passes either gate.

File: agents/murdock.md [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+# the in-flight-cancel test gap (WI-207/WI-208, 5 runs/arm) found no quality
+# difference — both arms 0/5 on both scenarios — at ~1.9x the cost ($5.12 vs
+# $2.70). Murdock was also the most expensive agent in M-20260703-001 ($26.09 of
+# $97.23). Caveat: both arms scored zero, so this is a floor result — it shows
+# opus bought nothing on that gap, not that the models are equivalent generally.
+model: sonnet
+# raised low -> high with the model change: test design is the reasoning-heavy
+# part of Murdock's job (deriving edge cases the ACs don't spell out), so buy
+# thinking on the cheaper model rather than stacking two reductions at once.
+effort: high
```
[the remark this answers is not in the document]
Comment (queso) [reply agents/murdock.md:13]: Fixed in `c29934b`, though not quite as suggested.

Swapping only the model name would have left "— test design quality is the load-bearing step for the rest of the pipeline" attached to `sonnet`. That clause was the argument *for* opus, so on sonnet it reads as a non-sequitur — arguably an argument against the change that was just made.

Every other agent's `## Model` section is a bare model name (amy/ba/lynch/tawnia: `sonnet`; face/sosa/stockwell: `opus`), none carrying rationale. Murdock's section is now just `sonnet`, matching that convention.

The rationale — the opus-5 vs sonnet-5 A/B (no measured quality difference, ~1.9x cost), and the caveat that both arms scored zero so it's a floor result — lives in the frontmatter comment next to `model:`. Keeping it in one place is the point: the duplicated copy in the body is precisely what drifted and produced this contradiction.

File: commands/plan.md [hunk 3]
```diff
@@ -266,8 +268,12 @@ SendMessage(
   1. Run ateam deps-check checkDeps --json to get the readyItems list
   2. Move items with NO dependencies to ready stage using ateam board-move moveItem
   3. Leave items WITH dependencies in briefings stage
+  4. Record ADRs: if Sosa's report has a non-empty 'ADR Candidates' section,
+     write each as adr/NNNN-*.md in the target repo per your ADR Recording
+     procedure (Glob for the next number, Write the file). State the ADR
+     outcome explicitly — files written, or exactly 'ADR Candidates: none.'
 
-  Report what was updated and moved."
+  Report what was updated and moved, including the ADR outcome."
```
[the remark this answers is not in the document]
Comment (queso) [reply commands/plan.md]: Fixed in `c29934b`. Correct catch — Sosa's report *always* contains an `ADR Candidates` section (it's mandatory in `agents/sosa.md`), so "non-empty" was never the right test and could have had Face record an ADR for a no-candidate report.

All three sites now key off the sentinel — the SendMessage second pass, the fresh-agent tool carve-out, and the fresh-agent step list:

> if Sosa's 'ADR Candidates' section is anything other than exactly 'ADR Candidates: none.' (the section is ALWAYS present, so its mere existence is not the signal — the sentinel is)

The parenthetical is there deliberately, since "the section exists" is the intuitive-but-wrong reading.

File: skills/test-writing/SKILL.md [hunk 4]
```diff
@@ -842,6 +842,27 @@ it('drops repeated submits while a request is in flight', async () => {
 });
 ```
 
+**Repeating the action is only half of it — also test the OPPOSING action during the same in-flight window.** A guard that drops repeat submits does nothing about the user who *cancels* mid-flight. Whenever an async action can be started, ask what else the UI still lets the user do before it settles — cancel, Escape, dismiss, close the dialog, navigate away — and test that path too. The failure is silent and specific: the abandoned action's promise still settles and applies its result, re-applying work the user believed they had discarded (or discarding work they believed they had kept).
```
[the remark this answers is not in the document]
Comment (queso) [reply skills/test-writing/SKILL.md]: Fixed in `c29934b`. `abort` now appears in both places, consistent with the trigger list at line 586:

- opposing-action list: "cancel, Escape, dismiss, abort, close the dialog, navigate away"
- per-control coverage rule: "the Escape key, an explicit abort, the close affordance, unmounting"

File: skills/test-writing/SKILL.md [hunk 5]
```diff
@@ -842,6 +842,27 @@ it('drops repeated submits while a request is in flight', async () => {
...
+it('does not reapply the edit when Cancel is clicked while the save is in flight', async () => {
+  let resolveSave!: (t: Todo) => void;
+  vi.mocked(updateTodo).mockReturnValue(new Promise((r) => { resolveSave = r; }));
+  render(<TodoItem todo={todo} onChanged={onChanged} {...rest} />);
+  await user.click(screen.getByRole('button', { name: /edit/i }));
+  await user.type(screen.getByRole('textbox'), 'Abandoned edit');
+  await user.keyboard('{Enter}');                                  // save now in flight
+  await user.click(screen.getByRole('button', { name: /cancel/i })); // opposing action
+  resolveSave({ ...todo, title: 'Abandoned edit' });               // settles AFTER the cancel
+  await waitFor(() => expect(onChanged).not.toHaveBeenCalled());
```
[the remark this answers is not in the document]
Comment (queso) [reply skills/test-writing/SKILL.md:865]: Fixed in `c29934b`. Good catch, and a pointed one — this is an exemplar inside the skill that bans vacuous tests, so it should not have been able to pass for the wrong reason. Agents copy exemplars literally, which makes it worse than an ordinary weak test.

Added the call assertion before the cancel, and switched to `mockImplementation` so each call gets its own promise rather than sharing one:

```typescript
vi.mocked(updateTodo).mockImplementation(
  () => new Promise((resolve) => { resolveSave = resolve; }),
);
...
await user.keyboard('{Enter}');                                  // save now in flight
// Prove the save actually started, or the assertion below passes vacuously.
await waitFor(() => expect(updateTodo).toHaveBeenCalledTimes(1));
await user.click(screen.getByRole('button', { name: /cancel/i })); // opposing action
resolveSave({ ...todo, title: 'Abandoned edit' });               // settles AFTER the cancel
await waitFor(() => expect(onChanged).not.toHaveBeenCalled());
```
